lib/travis/api/app/endpoint/authorization.rb

```diff
@@ -160,6 +162,8 @@ def handshake
           response.set_cookie('travis.state', value: state, httponly: true)
           redirect to(vcs_data['authorize_url'])
         end
+      rescue ::Travis::RemoteVCS::ConnectionError
+        halt 401, "Can't login"
```

Comment:
Should we redirect user to some user-friendly error page?

Reply:
We don't have it currently, we can add it as addition but is not strict related to this change and should be solved in different ticket (with web page team support)